src/NuGetGallery/Controllers/UsersController.cs

```
@@ -462,62 +463,128 @@ public virtual ActionResult Thanks()
+                OwnerRequests = GetOwnerRequestsForUser(currentUser),
+                ReservedNamespaces = GetReservedNamespacesForUser(currentUser),
+                WasMultiFactorAuthenticated = User.WasMultiFactorAuthenticated(),
+                IsCertificatesUIEnabled = ContentObjectService.CertificatesConfiguration?.IsUIEnabledForUser(currentUser) ?? false
+            };
 
-            var owners = new List<ListPackageOwnerViewModel> {
-                new ListPackageOwnerViewModel
+            return View(model);
+        }
+
+        [HttpGet]
+        [UIAuthorize]
+        public virtual JsonResult PackagesPaged(bool listed, int page = 0, string username = null)
+        {
+            return Json(
+                GetPagedPackagesForManagePackages(listed, page, username),
+                JsonRequestBehavior.AllowGet);
+        }
+
+        private dynamic GetPagedPackagesForManagePackages(bool listed, int page = 0, string username = null)
+        {
+            var currentUser = GetCurrentUser();
+            IEnumerable<Package> packages;
+            if (username == null)
+            {
+                packages = PackageService.FindPackagesByAnyMatchingOwner(
+                    currentUser, includeUnlisted: true, includeVersions: true);
+            }
+            else
+            {
+                var user = UserService.FindByUsername(username);
+                if (user != null)
+                {
+                    packages = PackageService.FindPackagesByOwner(
+                        user, includeUnlisted: true, includeVersions: true);
+                }
+                else
                 {
-                    Username = "All packages"
-                },
-                new ListPackageOwnerViewModel(currentUser)
-            }.Concat(currentUser.Organizations.Select(o => new ListPackageOwnerViewModel(o.Organization)));
+                    packages = Enumerable.Empty<Package>();
```

>packages [](start = 20, length = 8)

When would this case occur? Why not return an error code?

---

I answered this question above for @chenriksson, but to reiterate

This can occur when:
1. there is a bug in the page that causes it to request an invalid user
2. someone hacks the page to request a user that doesn't exist
3. an organization that the user was a member of was deleted after the page loaded

After thinking about it again, I think returning an error code might be useful so that the user can tell the page is having issues.